Store.js

```diff
@@ -140,14 +151,168 @@ define(["dcl/dcl", "decor/Invalidating"], function (dcl, Invalidating) {
 						collection.on("delete", this._itemRemoved.bind(this));
 						collection.on("refresh", this._refreshHandler.bind(this));
 					}
-					return this.processCollection(collection);
+				} else {
+					for (var i = 0; i < this.store.length; i++) {
+						if (this._isItemObservable(this.store[i])) {
+							this._hasItemObservable = true;
+							this._observeResults =  [];
```

Comment:
Do you really want to reset `this._observeResults` every occurrence in the loop?